staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -810,6 +818,16 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 			req = req.WithContext(r.ctx)
 		}
 		req.Header = r.headers
+		span := opentracing.SpanFromContext(r.ctx)
+		if span != nil {
+			ext.HTTPMethod.Set(span, req.Method)
+			ext.HTTPUrl.Set(span, req.URL.String())
+			ext.SpanKindRPCClient.Set(span)
+			opentracing.GlobalTracer().Inject(
+				span.Context(),
+				opentracing.HTTPHeaders,
+				opentracing.HTTPHeadersCarrier(req.Header))
+		}
 
 		r.backoffMgr.Sleep(r.backoffMgr.CalculateBackoff(r.URL()))
 		if retries > 0 {
```

Comment:
what will happen if there are retries?  The span seems to be created earlier, in the caller.